lib/format.js

```diff
@@ -152,6 +152,10 @@ format = function format(theme, options = {}) {
         theme.results.error = _.orderBy(theme.results.error, 'fatal', 'desc');
     }
 
+    if (theme.results.error.length && options.onlyFatalErrors || theme.results.hasFatalErrors) {
```

Comment:
This seems like not an optimal place to be handling file removal because the function itself should just be doing error formatting. Deleting files seems like a bit too much of a side effect :smiley:  

We also need to be safer with what we delete and allow for optional file removal. Would suggest moving this logic a layer up to https://github.com/TryGhost/gscan/blob/master/lib/index.js#L46, and create a wrapper function with  option similar to `keepExtractedDir` in [here](https://github.com/TryGhost/gscan/blob/master/lib/index.js#L27). This would give us bit more separation of concerns and configurability :+1:

Reply:
> Would suggest moving this logic a layer up to https://github.com/TryGhost/gscan/blob/master/lib/index.js#L46, and create a wrapper function with option similar to keepExtractedDir in here.

The funny thing is that I first thought of doing something similar as your suggestion, but thought that it wasn't the right approach. Now I see that it clearly is, as you say, format.js is just for error formatting. At the end I just searched for the smallest way to fix it, so at the Ghost Core there wouldn't be that much to change. 
So thanks for the feedback, I had some busy days, but tomorrow I'll be able to work on it 👍 
The only question I have for now, is that the optional file removal should be enabled by default, right?